src/capi/TuriCreate.h

```diff
@@ -0,0 +1,861 @@
+tc_sarray* tc_variant_sarray(const tc_variant*, tc_error**);
+tc_sframe* tc_variant_sframe(const tc_variant*, tc_error**);
+tc_parameters* tc_variant_parameters(const tc_variant*, tc_error** error);
+tc_model* tc_variant_model(const tc_variant*, tc_error**);
+
+
+/******************************************************************************/
+/*                                                                            */
+/*   Parameter Specification                                                  */
+/*                                                                            */
+/******************************************************************************/
+
+// A parameter specification is simply a map of names to variant types holding the
+// possible parameters.
+
+// Primary methods.
+tc_parameters* tc_parameters_create_empty(tc_error**);
+void tc_parameters_add(tc_parameters*, const char* name, const tc_variant*, tc_error** error);
+bool tc_parameters_entry_exists(const tc_parameters*, const char* name, tc_error** error);
+tc_variant* tc_parameters_retrieve(const tc_parameters*, const char* name, tc_error** error);
+
+// Convenience methods -- these can be expressed as combinations of the above methods,
+// but are provided here for convenience and to avoid the additional overhead of multiple function calls.
+void tc_parameters_add_int64(tc_parameters*, const char* name, int64_t value, tc_error** error);
+void tc_parameters_add_double(tc_parameters*, const char* name, double value, tc_error** error);
+void tc_parameters_add_cstring(tc_parameters*, const char* name, const char* str, tc_error** error);
+void tc_parameters_add_string(tc_parameters*, const char* name, const char* str, uint64_t n, tc_error** error);
+void tc_parameters_add_double_array(tc_parameters*, const char* name, const double* data, uint64_t n, tc_error** error);
+void tc_parameters_add_flex_list(tc_parameters*, const char* name, const tc_flex_list* value, tc_error** error);
+void tc_parameters_add_flex_dict(tc_parameters*, const char* name, const tc_flex_dict* value, tc_error** error);
+void tc_parameters_add_datetime(tc_parameters*, const char* name, const tc_datetime* dt, tc_error**);
+void tc_parameters_add_image(tc_parameters*, const char* name, const tc_flex_image*, tc_error** error);
+void tc_parameters_add_flexible_type(tc_parameters*, const char* name, const tc_flexible_type*, tc_error** error);
+void tc_parameters_add_sarray(tc_parameters*, const char* name, const tc_sarray*, tc_error** error);
+void tc_parameters_add_sframe(tc_parameters*, const char* name, const tc_sframe*, tc_error** error);
+void tc_parameters_add_parameters(tc_parameters*, const char* name, const tc_parameters*, tc_error** error);
+void tc_parameters_add_model(tc_parameters*, const char* name, const tc_model*, tc_error** error);
+
+bool tc_parameters_is_int64(const tc_parameters*, const char* name, tc_error** error);
```

Comment:
Do you want to use ints here? There are previous places in this file where a bool would have been appropriate but an int was used instead?